src/components/RegisterForm.vue

```diff
@@ -0,0 +1,121 @@
+<template>
+  <v-dialog max-width="400px" v-model="dialog">
+    <template v-slot:activator="{ on }">
+      <v-btn color="#FF9F1C" v-on="on" @click="clearErrors">
+        <span class="d-none d-sm-flex ">Register</span>
+        <v-icon class="d-sm-none">mdi-account-plus-outline</v-icon>
+      </v-btn>
+    </template>
+    <v-card color="#FFBF69">
+      <v-card-title class="justify-center">
+        <h2>Register</h2>
+      </v-card-title>
+      <v-card-text>
+        <v-alert class="mt-4" v-if="serverErrors" type="error">
+          <div v-for="(value, key) in serverErrors" :key="key">
+            {{ value[0] }}
+          </div>
+        </v-alert>
+        <v-form v-model="valid" @submit.prevent="register">
+          <v-text-field
+            type="string"
+            label="Name"
+            v-model="name"
+            prepend-icon="mdi-account"
+            :rules="nameRules"
+            required
+          ></v-text-field>
+          <v-text-field
+            type="email"
+            label="Email"
+            v-model="email"
+            prepend-icon="mdi-account-circle"
+            :rules="emailRules"
+            required
+          ></v-text-field>
+          <v-text-field
+            :type="showPassword ? 'text' : 'password'"
+            label="password"
+            v-model="password"
+            prepend-icon="mdi-lock"
+            :append-icon="showPassword ? 'mdi-eye' : 'mdi-eye-off'"
+            @click:append="showPassword = !showPassword"
+            :rules="passwordRules"
+            required
+          ></v-text-field>
+          <v-text-field
+            :type="showPassword ? 'text' : 'password'"
+            label="re_password"
+            v-model="re_password"
+            prepend-icon="mdi-lock"
+            :append-icon="showPassword ? 'mdi-eye' : 'mdi-eye-off'"
+            @click:append="showPassword = !showPassword"
+            :rules="passwordRules"
+            required
+          ></v-text-field>
+          <v-btn color="#FF9F1C" type="submit" :loading="loading"
+            >Register</v-btn
+          >
+        </v-form>
+      </v-card-text>
+    </v-card>
+  </v-dialog>
+</template>
+
+<script>
+export default {
+  name: "RegisterForm",
+  data() {
+    return {
+      name: "",
+      email: "",
+      password: "",
+      re_password: "",
+      dialog: false,
+      loading: false,
+      showPassword: false,
+      serverErrors: "",
+      valid: false,
+      emailRules: [
+        v => !!v || "E-mail is required",
+        v => /.+@.+\..+/.test(v) || "This is not correct e-mail"
+      ],
+      passwordRules: [
+        v => !!v || "Password is required",
+        v => v.length > 8 || "Password need to have at least 8 letters",
+        v => this.password === v || "Passwords need to be the same"
+      ],
+      nameRules: [
+        v => v.length > 3 || "Name is to short",
+        v => v.length < 20 || "Name is to long"
+      ]
```

Comment:
maybe is a good way is to use vee-validate instead of writing your own rules on each form? As far as i remember, vuetify supports and implies defaults vee-validate as a validator

Reply:
Or maybe just use validation messages from backend?

Reply:
I think frontend validation would be a nice thing to have. It will make forms more responsive for end user.

Reply:
I do not think it is necessary to use vee-validate. 
There are messages from backend too but as Szymon said using frontend validation also give feeling of responsibility to end user

Reply:
@HerringTheCoder, what about DRY? ;)
In this case change of validation would require at least two changes: on backend and on frontend. Forms could be more responsive even with sending requests to the backend.

Reply:
I think both approaches have their merits and would depend on the data we are talking about. Backend validation is a must have, obviously, but frontend validation rules are usually much simpler (name length, mail regex).

This type of client-side validation also decreases load from API and allows for instant rule checks against individual keypresses and that alone is a separate client-side feature, so I think we could justify a slight repetition here.

Still, if there is some other clever way to do that, then I'm all ears.

Reply:
From my point of view, validation on frontend side, is nice to have, because basically f.e it restricts sending an empty requests. Like @HerringTheCoder  said basic rules like required and email don't change often and isn't something hard to do.

@KaSliK If we back to topic, its better way to have all rules in one file instead of having different rules for each form( it breaks DRY rule?)